### PR TITLE
Add X-Robots-Tag header to response for items marked as no_index

### PR DESCRIPTION
**Pull request by sneridagh**

### CODE REVIEW

File: src/kitconcept/seo/events.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from ZPublisher.interfaces import IPubAfterTraversal
+from ZPublisher.pubevents import PubAfterTraversal
+
+
+@adapter(IPubAfterTraversal)
+def after_image_pub_traversal(event: PubAfterTraversal):
+    """Record the view as the HTTP route after traversal"""
+    request = event.request
+    context = event.request["PARENTS"][0]
+    if IImage.providedBy(context) and context.seo_noindex:
```
Comment (davisagli) [inline src/kitconcept/seo/events.py:12]: This needs to handle objects that provide IImage but don't have the SEO behavior enabled.

I'd also like a test for this feature.

**Comment by davisagli** [comment]
@sneridagh I updated this to add the header for any item that has the behavior enabled and seo_noindex checked (not just images).

But I tested it in our project and I think it needs more work:
1. The header is added to backend responses, but Volto's proxy does not pass it through.
2. The header is added for the default view of the item, but not for related paths like ++api++ traversal or downloading image scales. I think we want to add it if an item with seo_noindex is the closest content item in request["PARENTS"], even if it's not the first item.
3. This gets a bit more complicated, so I think some functional tests would be good.

**Comment by sneridagh** [comment]
@davisagli 
1. fixed and released: https://github.com/plone/volto/releases/tag/16.31.9
2. Why do we need the header in `++api++`? the bots won't never search in there, nor parse the JSON in search for images.

**Comment by davisagli** [comment]
@sneridagh Are you sure? In the logs from our project, I see traffic from Bingbot on API URLs.

**Comment by davisagli** [comment]
@sneridagh Anyway, we definitely need it for image scales.

**Review by davisagli — APPROVED** [review]

**Pull request merged**